src/pm.c

```diff
@@ -0,0 +1,472 @@
+#include <stdarg.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#define PM_COUNTER	"counter"
+#define PM_GAUGE	"gauge"
+#define PM_HISTOGRAM	"histogram"
+
+#define PM_LEN			128
+#define PM_LABEL_MAX		5
+#define PM_LABEL_VAL_MAX	5
+#define PM_METRIC_MAX		30
+#define PM_HIST_BIN_MAX		10
+
+typedef struct {
+	char		name[PM_LEN];
+	char		help[PM_LEN];
+	char		type[PM_LEN];
+	char		label[PM_LABEL_MAX][PM_LEN];
+	uint32_t	n_label;
+	double		bin[PM_HIST_BIN_MAX];
+	uint32_t	n_bin;
+	struct {
+		char	lblval[PM_LABEL_MAX][PM_LEN];
+		union {
+			double	 	counter;
+			double		gauge;
+			struct {
+				double		sum;
+				uint64_t	cnt[PM_HIST_BIN_MAX];
+				uint64_t	inf;
+			} histogram;
+		};
+	} s[PM_LABEL_VAL_MAX];
+	uint32_t	n_store;
+} pm_metric;
+
+static int		 pm_metric_get_store(pm_metric *m, va_list *);
+static pm_metric	 metrics[PM_METRIC_MAX] = {{{0}}};
+
+int
+pm_metric_get_store(pm_metric *m, va_list *ap1)
```

Comment:
This function is a little hard to read - mind adding some comments here to help the reader understand what is happening?